core/src/main/java/org/apache/accumulo/core/data/Condition.java

```diff
@@ -308,41 +309,20 @@ public Condition setIterators(IteratorSetting... iterators) {
 
   @Override
   public boolean equals(Object o) {
-    if (o == this) {
+    if (this == o)
       return true;
-    }
-    if (o == null || !(o instanceof Condition)) {
-      return false;
-    }
-    Condition c = (Condition) o;
-    if (!(c.cf.equals(cf))) {
-      return false;
-    }
-    if (!(c.cq.equals(cq))) {
-      return false;
-    }
-    if (!(c.cv.equals(cv))) {
+    if (o == null || !(o instanceof Condition))
```

Comment:
This could be simplified to:

```suggestion
    if (!(o instanceof Condition))
```